core/src/apps/bitcoin/sign_tx/zcash.py

```diff
@@ -90,69 +169,8 @@ def write_tx_footer(self, w: Writer, tx: Union[SignTx, PrevTx]) -> None:
         if tx.version >= 3:
             write_uint32(w, tx.expiry)  # expiryHeight
 
-    # ZIP-0143 / ZIP-0243
-    # ===
-
-    def init_hash143(self) -> None:
-        self.h_prevouts = HashWriter(blake2b(outlen=32, personal=b"ZcashPrevoutHash"))
-        self.h_sequence = HashWriter(blake2b(outlen=32, personal=b"ZcashSequencHash"))
-        self.h_outputs = HashWriter(blake2b(outlen=32, personal=b"ZcashOutputsHash"))
-
-    def hash143_preimage_hash(
-        self, txi: TxInput, public_keys: List[bytes], threshold: int
-    ) -> bytes:
-        h_preimage = HashWriter(
-            blake2b(
-                outlen=32,
-                personal=b"ZcashSigHash" + struct.pack("<I", self.tx.branch_id),
-            )
-        )
-
-        assert self.tx.version_group_id is not None
-        assert self.tx.expiry is not None
-
-        # 1. nVersion | fOverwintered
-        write_uint32(h_preimage, self.tx.version | OVERWINTERED)
-        # 2. nVersionGroupId
-        write_uint32(h_preimage, self.tx.version_group_id)
-        # 3. hashPrevouts
-        write_bytes_fixed(h_preimage, get_tx_hash(self.h_prevouts), TX_HASH_SIZE)
-        # 4. hashSequence
-        write_bytes_fixed(h_preimage, get_tx_hash(self.h_sequence), TX_HASH_SIZE)
-        # 5. hashOutputs
-        write_bytes_fixed(h_preimage, get_tx_hash(self.h_outputs), TX_HASH_SIZE)
-
-        zero_hash = b"\x00" * TX_HASH_SIZE
-        # 6. hashJoinSplits
-        write_bytes_fixed(h_preimage, zero_hash, TX_HASH_SIZE)
-        # 7. hashShieldedSpends
-        write_bytes_fixed(h_preimage, zero_hash, TX_HASH_SIZE)
-        # 8. hashShieldedOutputs
-        write_bytes_fixed(h_preimage, zero_hash, TX_HASH_SIZE)
-        # 9. nLockTime
-        write_uint32(h_preimage, self.tx.lock_time)
-        # 10. expiryHeight
-        write_uint32(h_preimage, self.tx.expiry)
-        # 11. valueBalance
-        write_uint64(h_preimage, 0)
-        # 12. nHashType
-        write_uint32(h_preimage, self.get_sighash_type(txi))
-
-        # 10a /13a. outpoint
-        write_bytes_reversed(h_preimage, txi.prev_hash, TX_HASH_SIZE)
-        write_uint32(h_preimage, txi.prev_index)
-
-        # 10b / 13b. scriptCode
-        script_code = derive_script_code(txi, public_keys, threshold, self.coin)
-        write_bytes_prefixed(h_preimage, script_code)
-
-        # 10c / 13c. value
-        write_uint64(h_preimage, txi.amount)
-
-        # 10d / 13d. nSequence
-        write_uint32(h_preimage, txi.sequence)
-
-        return get_tx_hash(h_preimage)
+        assert self.tx_info.tx.version_group_id is not None
+        assert self.tx_info.tx.expiry is not None
```

Comment:
these assertions don't seem to do anything here?

Reply:
Yeah, really weird. Probably got introduced during some rebase, because these are originally the asserts which are now on lines 62 and 63.

Reply:
https://github.com/trezor/trezor-firmware/pull/1292/commits/66a13481fbe78bd141585cb92e4ff149c9e2021c